app/webpacker/components/Delegates/DelegatesTable.jsx

```diff
@@ -1,60 +1,68 @@
-import React, { useMemo } from 'react';
+import React, { useMemo, useReducer } from 'react';
 import { Button, Table } from 'semantic-ui-react';
 import _ from 'lodash';
 import cn from 'classnames';
 import I18n from '../../lib/i18n';
 import { competitionsUrl } from '../../lib/requests/routes.js.erb';
 import UserBadge from '../UserBadge';
+import createSortReducer from '../../lib/reducers/sortReducer';
 
 const dasherize = (string) => _.kebabCase(string);
 
+const columns = ['name', 'role', 'region', 'first_delegated', 'last_delegated', 'total_delegated'];
+
+const sortReducer = createSortReducer(columns);
+
 export default function DelegatesTable({
-  delegates, isAdminMode, isAllLeadDelegates, isAllNonLeadDelegates,
+  delegates,
+  isAdminMode,
+  isAllLeadDelegates,
+  isAllNonLeadDelegates,
 }) {
-  const tableData = useMemo(() => delegates.filter(
-    (delegate) => delegate.metadata.status !== 'trainee_delegate' || isAdminMode,
-  ), [delegates, isAdminMode]);
+  const [sortedData, dispatch] = useReducer(sortReducer, {
```

Comment:
I'd rename `sortedData` to `sortState`, or even just use `{ sortColumn, sortDirection }` directly to avoid naming it.